app/src/main/java/com/google/samples/apps/sunflower/compose/Insets.kt

```diff
@@ -0,0 +1,136 @@
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.compose
+
+import android.util.Log
+import androidx.compose.Composable
+import androidx.compose.Immutable
+import androidx.compose.Providers
+import androidx.compose.ambientOf
+import androidx.compose.getValue
+import androidx.compose.onCommit
+import androidx.compose.setValue
+import androidx.compose.state
+import androidx.core.graphics.Insets
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.doOnAttach
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.core.ViewAmbient
+import androidx.ui.core.composed
+import androidx.ui.geometry.Rect
+import androidx.ui.layout.absolutePadding
+import androidx.ui.unit.Density
+import androidx.ui.unit.Dp
+import androidx.ui.unit.dp
+
+// Code taken from github.com/chrisbanes/tivi
+/**
+ * Main holder of our inset values.
+ */
+data class ComposeInsets(
+    val systemBars: Rect = Rect.zero
+)
+
+val InsetsAmbient = ambientOf { ComposeInsets() }
+
+@Composable
+fun ProvideInsets(children: @Composable () -> Unit) {
+    val view = ViewAmbient.current
+    var insetsHolder by state { ComposeInsets() }
+    Log.e("TEST", insetsHolder.toString())
+
+    onCommit(view.id) {
+        Log.e("TEST", "On Commit called")
+        ViewCompat.setOnApplyWindowInsetsListener(view) { _, insets ->
+            insetsHolder = insets.toComposeInsets()
+            Log.e("TEST", insetsHolder.toString())
+            // Return the unconsumed insets
+            insets
+        }
+
+        view.doOnAttach {
+            ViewCompat.requestApplyInsets(it)
+        }
+
+        onDispose {
+            ViewCompat.setOnApplyWindowInsetsListener(view, null)
+        }
+    }
+
+    Providers(InsetsAmbient provides insetsHolder, children = children)
+}
+
+fun Modifier.systemBarsPadding(all: Boolean = false) = systemBarsPadding(all, all, all, all)
+
+fun Modifier.systemBarsPadding(
+    left: Boolean = true,
+    top: Boolean = true,
+    right: Boolean = true,
+    bottom: Boolean = true
+) = composed {
+    val systemBars = InsetsAmbient.current.systemBars
+    val padding = systemBars.toAbsoluteInnerPadding(DensityAmbient.current)
+    absolutePadding(padding, left, top, right, bottom)
+}
+
+/**
+ * Allows conditional setting of [padding] on each dimension.
+ */
+fun Modifier.absolutePadding(
```

Comment:
Should this be private? Doesn't look like you'd want to call this outside of here.